anvil/src/input_handler.rs

```diff
@@ -242,47 +257,16 @@ impl AnvilState<UdevData> {
     }
 
     fn clamp_coords(&self, pos: (f64, f64)) -> (f64, f64) {
-        if self.backend_data.output_map.is_empty() {
+        if self.output_map.borrow().is_empty() {
             return pos;
         }
 
-        let (mut x, mut y) = pos;
-        // max_x is the sum of the width of all outputs
-        let max_x = self
-            .backend_data
-            .output_map
-            .iter()
-            .fold(0u32, |acc, output| acc + output.size.0);
-        x = x.max(0.0).min(max_x as f64);
-
-        // max y depends on the current output
-        let max_y = self.current_output_size(x).1;
-        y = y.max(0.0).min(max_y as f64);
-
-        (x, y)
-    }
-
-    fn current_output_idx(&self, x: f64) -> usize {
-        self.backend_data
-            .output_map
-            .iter()
-            // map each output to their x position
-            .scan(0u32, |acc, output| {
-                let curr_x = *acc;
-                *acc += output.size.0;
-                Some(curr_x)
-            })
-            // get an index
-            .enumerate()
-            // find the first one with a greater x
-            .find(|(_idx, x_pos)| *x_pos as f64 > x)
-            // the previous output is the one we are on
-            .map(|(idx, _)| idx - 1)
-            .unwrap_or(self.backend_data.output_map.len() - 1)
-    }
+        let (pos_x, pos_y) = pos;
+        let (max_x, max_y) = self.output_map.borrow().size();
+        let clamped_x = pos_x.max(0.0).min(max_x as f64);
+        let clamped_y = pos_y.max(0.0).min(max_y as f64);
```

Comment:
Isn't this a little too simplistic? Like this the cursor can get stuck in an invisible area between two very different monitors (e.g. a horizontal and a vertical one), no?

Reply:
Should not be the case, at the moment the overall size calculated by the output_map is the sum of all widths and the maximum height of the outputs. This is similiar to what was implemented before. I agree that is will get more complicated when the outputs can be placed differently. I left a note in the `output_map` that the current implementation is really simple.

Reply:
So if two outputs have very different heights, what happens if I move the cursor one the smaller screen below its height? Should still be inside the max-height, right?

Reply:
Ah, now i know what you mean, sorry for the misunderstanding. I thought you mean it will get stuck like unable to move. Yes, with this simple implementation the cursor could be outside of a screen output if multiple monitors with different resolutions are used. This was intentional, but if it is an issue i can implement output specific clamping. Personally i hate it when i am not able to move the cursor from one screen to another cause their heights are different and I have to move the cursor up before. An alternative could be  to set the cursor to the max height of the next output when moving on the x-axis. Any preference?

Reply:
Ok, I think now the logic should be how it was before I changed it and the height is clamped with the current output height of the cursor.

Reply:
Fixed the wrong clamping, but only tested on a two monitor setup. But with two different resolutions.